plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php

```diff
@@ -86,6 +90,71 @@ protected function initialize() {
 		add_filter( 'render_block_data', array( $this, 'disable_enhanced_pagination' ), 10, 1 );
 	}
 
+	/**
+	 * Provides the location context to each inner block of the product collection block.
+	 * Hint: Only blocks using the 'query' context will be affected.
+	 *
+	 * The sourceData structure depends on the context type as follows:
+	 * - site:    [ ]
+	 * - order:   [ 'orderId'    => int ]
+	 * - cart:    [ 'productIds' => int[] ]
+	 * - archive: [ 'taxonomy'   => string, 'termId' => int ]
+	 * - product: [ 'productId'  => int ]
+	 *
+	 * @example array(
+	 *   'type'       => 'product',
+	 *   'sourceData' => array( 'productId' => 123 ),
+	 * )
+	 *
+	 * @param array $context  The block context.
+	 * @return array $context {
+	 *     The block context including the product collection location context.
+	 *
+	 *     @type array $productCollectionLocation {
+	 *         @type string  $type        The context type. Possible values are 'site', 'order', 'cart', 'archive', 'product'.
+	 *         @type array   $sourceData  The context source data. Can be the product ID of the viewed product, the order ID of the current order viewed, etc. See structure above for more details.
+	 *     }
+	 * }
+	 */
+	public function provide_location_context_for_inner_blocks( $context ) {
+		// Run only on frontend.
+		// This is needed to avoid SSR renders while in editor. @see https://github.com/woocommerce/woocommerce/issues/45181.
+		if ( is_admin() || \WC()->is_rest_api_request() ) {
+			return $context;
+		}
+
+		// Target only product collection's inner blocks that use the 'query' context.
+		if ( ! isset( $context['query'] ) || ! isset( $context['query']['isProductCollectionBlock'] ) || ! $context['query']['isProductCollectionBlock'] ) {
+			return $context;
+		}
```

Comment:
Nice! I like how you are targeting inner blocks that use the 'query' context. In the future, we may want to think if we wanna add context even for blocks that don't use 'query' context, which won't be straightforward to implement. This looks good for now 🚀

Reply:
Good question! 

The "location" context is primarily used to help build the query -that's why this strong connection. 

Since I'm not certain if there will be a need for this location beyond the query, we can keep it as it is. We can revisit this if there is a demand and if there are specific use cases that justify the added complexity!